rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,159 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
+                assert_eq!(r.len(), 0);
+                assert_eq!(res.value, buf);
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_lookup() {
+
+        let buf: &[u8] = &[
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+        // [name]
+            0x00, 0x00, 0x00, 0x02, 0x61, 0x6d, 0x00, 0x00,
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+        assert_eq!(handle.value, &buf[..32]);
+
+        let result = parse_nfs2_request_lookup(buf).unwrap();
+        match result {
+            (r, request) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.name_vec, b"am".to_vec());
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_read() {
+
+        let buf: &[u8] = &[
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7, // [file_handle]
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5d,
+            0x00, 0x00, 0x00, 0x2a, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+            0x00, 0x00, 0x00, 0x00,                         // offset
+            0x00, 0x00, 0x20, 0x00,                         // count
+            0x00, 0x00, 0x20, 0x00,                         // total count
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+        assert_eq!(handle.value, &buf[..32]);
+
+        let result = parse_nfs2_request_read(buf).unwrap();
+        match result {
+            (r, request) => {
+                assert_eq!(r.len(), 4);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.offset, 0);
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_reply_read() {
+
+        let buf: &[u8] = &[
+        // Status: NFS_OK - (0)
+            0x00, 0x00, 0x00, 0x00,
+        // attr_blob
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x81, 0xa4,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x0b,
+            0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xb2, 0x5d, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x0b, 0x71, 0xb0, 0x38, 0x47, 0x71, 0xc4,
+            0x00, 0x08, 0xb2, 0x90, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x09, 0x00, 0xb0,
+        // data_len: (11)
+            0x00, 0x00, 0x00, 0x0b,
+        // data_contents: ("the b file")
+            0x74, 0x68, 0x65, 0x20,
+            0x62, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x0a,
+        // _data_padding: parsed as part of data_contents
+            0x00,
```

Comment:
I had to specify this `_data_padding` byte here since it confused me while testing against `data_contents` as it was parsed with it and the assertion failed due to mismatched sizes. 
`data_contents` in the parser is larger than its respective PCAP field with one byte.

Reply:
Could this be solved if the comment line was `// data_contents: ("the b file") + _data_padding`, or do you think that wouldn't be clear enough?

Reply:
The issue here is that the parser isn't correct. If you look at the pcap in wireshark you'll see the `fill bytes` logic. I suspect it fills bytes to make the payload a multiple of 4.

Reply:
https://github.com/OISF/suricata/pull/6764/commits/ddf14e51dcf5dabdb4bc07414a118530e2a1b819

Reply:
I've pushed ddf14e5 to master, can you do a new PR where you rebase your branch to master?

Reply:
Ok good, thanks!